src/appleseed/renderer/modeling/project/projectfilereader.cpp

```diff
@@ -3036,15 +3038,49 @@ namespace
     }
 }
 
+bool is_packed_project(const string& project_filepath, string& project_name) {
+    try 
+    {
+        vector<string> appleseed_files = get_filenames_with_extension_from_zip(project_filepath, ".appleseed");
+        
+        if (appleseed_files.size() != 1)
+            throw ExceptionUnsupportedFileFormat("Should be exactly one .appleseed file in project");
```

Comment:
These two are different exceptions.
UnzipException comes from `get_filenames_with_extension_from_zip` function in case of IO errors or if file is not a zip
ExceptionUnsupportedFileFormat throws to outside of the function